File: JavaScript/programmers/etc/영어끝말잇기.js
function solution(n, words) {
  const cache = { [words[0]]: true };
  let round = 1;
  let prev = words[0];
  for (let i = 1; i < words.length; i++) {
    if (i % n === 0) {
      round++;
    }
    if (cache[words[i]] || words[i][0] !== prev[prev.length - 1]) {
      return [(i % n) + 1, round];
    } else {
      cache[words[i]] = true;
      prev = words[i];
    }
  }
  return [0, 0];
}

File: JAVA/programmers/lv2/1217/_lv2/영어끝말잇기.java
package category.ns.programmersTMP._lv2;
import java.util.*;

public class 영어끝말잇기 {

    class Solution {
        public int[] solution(int n, String[] words) {
            Set<String> set = new HashSet<>();
            int[] answer = new int[2];

            String prev = null;
            for(int i = 0; i < words.length; i++){
                String cur = words[i];
                if(set.contains(cur) ||
                        ((prev != null) && (prev.charAt(prev.length()-1)) != cur.charAt(0))
                ){
                    answer[0] = i%n+1;
                    answer[1] = i/n+1;
                    break;
                }
                set.add(cur);
                prev = cur;
            }
            return answer;
        }
    }
}
